paymentsheet/src/main/java/com/stripe/android/paymentsheet/ui/PaymentMethodUI.kt

```diff
@@ -0,0 +1,219 @@
+    val iconSize = 16.dp
+}
+
+@VisibleForTesting
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+const val TEST_TAG_ICON_FROM_RES = "PaymentMethodsUIIconFomRes"
+
+@Composable
+internal fun PaymentMethodUI(
+    minViewWidth: Dp,
+    iconRes: Int,
+    iconUrl: String?,
+    imageLoader: StripeImageLoader,
+    title: String,
+    isSelected: Boolean,
+    isEnabled: Boolean,
+    iconRequiresTinting: Boolean,
+    modifier: Modifier = Modifier,
+    onItemSelectedListener: () -> Unit
+) {
+    RowButton(
+        isEnabled = isEnabled,
+        isSelected = isSelected,
+        onClick = onItemSelectedListener,
+        contentPaddingValues = PaddingValues(
+            start = PaymentMethodUISpacing.cardPadding,
+            end = PaymentMethodUISpacing.cardPadding,
+            top = PaymentMethodUISpacing.cardPadding,
+        ),
+        modifier = modifier
+            .height(60.dp)
+            .widthIn(min = minViewWidth),
+    ) {
+        PaymentMethodIconUi(
+            iconRes = iconRes,
+            iconUrl = iconUrl,
+            imageLoader = imageLoader,
+            iconRequiresTinting = iconRequiresTinting,
+            modifier = Modifier
+                .height(PaymentMethodUISpacing.iconSize),
+        )
+
+        LpmSelectorText(
+            text = title,
+            isEnabled = isEnabled,
+            textColor = MaterialTheme.stripeColors.onComponent,
+            modifier = Modifier.padding(top = 6.dp),
+        )
+    }
+}
+
+@Composable
+private fun PaymentMethodIconUi(
+    iconRes: Int,
+    iconUrl: String?,
+    imageLoader: StripeImageLoader,
+    iconRequiresTinting: Boolean,
+    modifier: Modifier,
+) {
+    val color = MaterialTheme.stripeColors.onComponent
+    val colorFilter = remember(iconRequiresTinting) {
+        if (iconRequiresTinting) {
+            ColorFilter.tint(color)
+        } else {
+            null
+        }
+    }
+
+    Box(
+        modifier = modifier,
+    ) {
+        if (iconUrl != null) {
+            StripeImage(
+                url = iconUrl,
+                imageLoader = imageLoader,
+                contentDescription = null,
+                contentScale = ContentScale.Fit,
+                errorContent = {
+                    PaymentMethodIconFromResource(
+                        iconRes = iconRes,
+                        colorFilter = colorFilter
+                    )
+                },
+            )
+        } else {
+            PaymentMethodIconFromResource(iconRes = iconRes, colorFilter = colorFilter)
+        }
+    }
+}
+
+@Composable
+private fun PaymentMethodIconFromResource(
+    iconRes: Int,
+    colorFilter: ColorFilter?,
+) {
+    if (iconRes != 0) {
+        Image(
+            painter = painterResource(iconRes),
+            contentDescription = null,
+            colorFilter = colorFilter,
+            modifier = Modifier.testTag(TEST_TAG_ICON_FROM_RES)
+        )
+    }
+}
+
+@Composable
+internal fun NewPaymentMethodRowButton(
```

Comment:
This is where the new content starts. This is a single row, similar to the above is a single tab.

Reply:
what do you think of renaming the existing one (PaymentMethodUI) to something that makes it clear it is a "tab"? Looking at this new one, it's obvious to me that this is a PM row but that's not clear with the old one. I don't think this is necessary for this PR, but would be a good follow up.

Less confident about this, but do you think that it would be better for discoverability to have these in separate files? e.g. have a `NewPaymentMethodRowButton` file and a `PaymentMethodTab` file or something like that. Might also help with the fact that right now PaymentMethodUI and PaymentMethodsUI are so similarly named

Reply:
Yeah, makes sense. I'll follow up.